### DISCUSSION TITLE
Delphi 2007 w/ Indy 10.6.2.0 Won't Compile (File not found: IdSys.dcu)

### DISCUSSION THREAD

**Discussion by millamdl** [IndySockets/Indy#607] [Q&A]
I've have a legacy application which is built with Delphi 2007 and sends emails using TIdSMTP component.  I needed to update it to support TLS 1.2 in order to send email via Amazon SES.  I updated the Indy Core and Protocols package to 10.6.20 but when I got to compile my application it fails because it is unable to find IdSys.dcu.  

Does anyone have an idea on where that is coming from?  Any help would be appreciated.

Thanks.

Dean Millam

**Comment by millamdl** [comment]
Figured out that I had some other package that was compiled using Indy components.

**Comment by rlebeau** [comment]
> I updated the Indy Core and Protocols package

There is 3 runtime packages - IndySystem, IndyCore, and IndyProtocols.  And 2 design-time packages - dclIndyCore and dclIndyProtocols.  All 5 need to be updated together, you can't just pick and choose which ones to update.

> when I got to compile my application it fails because it is unable to find IdSys.dcu.

Indy removed the `IdObjs` and `IdSys` units around 2007/2009.  Whatever code is looking for them is WAY outdated and needs to be updated to use contemporary Delphi RTL units instead.
